### PR TITLE
expression: implement vectorized evaluation for `builtinVersionSig`

### PR DESCRIPTION
**Pull request by TennyZhuang**
PCP #12106

### What problem does this PR solve? 

#12106

### What is changed and how it works?

vectorize `builtinVersionSig`

```
BenchmarkVectorizedBuiltinInfoFunc/builtinVersionSig-VecBuiltinFunc-4         	  126724	      9617 ns/op	       0 B/op	       0 allocs/op
BenchmarkVectorizedBuiltinInfoFunc/builtinVersionSig-NonVecBuiltinFunc-4      	   59805	     18713 ns/op	       0 B/op	       0 allocs/op
PASS
ok  	github.com/pingcap/tidb/expression	2.691s
```

### Check List 

Tests 

 - Unit test

Code changes

Side effects

Related changes

Release note

vectorize `builtinVersionSIg`

### CODE REVIEW

File: expression/builtin_info_vec.go [hunk 1]
```diff
@@ -137,11 +138,16 @@ func (b *builtinLastInsertIDWithIDSig) vecEvalInt(input *chunk.Chunk, result *ch
 }
 
 func (b *builtinVersionSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinVersionSig) vecEvalString(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
```
Comment (wshwsh12) [inline expression/builtin_info_vec.go]: I think we need a line `result.ReserveString(n)` to reset the result column.

File: expression/builtin_info_vec.go [hunk 2]
```diff
@@ -137,11 +138,16 @@ func (b *builtinLastInsertIDWithIDSig) vecEvalInt(input *chunk.Chunk, result *ch
...
+	for i := 0; i < n; i++ {
+		result.AppendString(mysql.ServerVersion)
+	}
+
```
Comment (wshwsh12) [inline expression/builtin_info_vec.go]: Useless line.

**Comment by wshwsh12** [comment]
And please add some test cases for your code.

**Comment by TennyZhuang** [comment]
@wshwsh12 

It seems that just add `ast.Version:     {},` in testcases is ok?

**Comment by wshwsh12** [comment]
@TennyZhuang   
```
ast.Version: {
		{retEvalType: types.ETString, childrenTypes: []types.EvalType{}},
	},
```

**Comment by TennyZhuang** [comment]
@SunRunAway I have update it in PR description.

**Pull request merged**